### PR TITLE
Phoenix/tree w cansee

### PR DESCRIPTION
**Pull request by phoenixAja**
### Description

restricting permission on viewable trees

#### Issue
[ch137589](https://app.clubhouse.io/genepi/story/137589)

### Test plan

Wrote some new tests

### CODE REVIEW

File: src/backend/aspen/app/views/phylo_trees.py [hunk 1]
```diff
@@ -36,6 +36,11 @@ def phylo_trees():
             .options(joinedload(User.group).joinedload(Group.can_see))
             .one()
         )
+        cansee_viewer_group_ids: Set[int] = {
+            cansee.viewer_group_id
+            for cansee in user.group.can_see
+            if cansee.data_type == DataType.TREES
+        }
         # TODO: Add subquery to fetch trees for which we have can-see permissions.
```
Comment (ttung) [inline src/backend/aspen/app/views/phylo_trees.py]: remove this pls.

**Review by ttung — APPROVED** [review]
perfect, except remove that one comment.

File: src/backend/aspen/app/views/phylo_trees.py [hunk 2]
```diff
@@ -36,6 +36,11 @@ def phylo_trees():
             .options(joinedload(User.group).joinedload(Group.can_see))
             .one()
         )
+        cansee_viewer_group_ids: Set[int] = {
+            cansee.viewer_group_id
```
Comment (ttung) [inline src/backend/aspen/app/views/phylo_trees.py]: oh, this should be cansee.owner_group_id

**Review by ttung — CHANGES_REQUESTED** [review]

File: src/backend/aspen/app/views/phylo_trees.py (hunk 2 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/phylo_trees.py]: cool i changed it! Thank you!!

File: src/backend/aspen/app/views/tests/test_phylo_tree_view.py [hunk 3]
```diff
@@ -1,46 +1,76 @@
...
+
+def make_trees(
+    group: Group, samples: Collection[Sample], n_trees: int
+) -> Collection[PhyloTree]:
+    # make up to n trees, each with a random sample of uploaded pathogen genomes.
+    return [
         phylotree_factory(
             phylorun_factory(group),
-            random.sample(samples, k=random.randint(0, len(samples))),
+            random.sample(samples, k=random.randint(0, len(samples))),  # type: ignore
```
Comment (ttung) [inline src/backend/aspen/app/views/tests/test_phylo_tree_view.py:48]: What happened here that is triggering the type checker?

**Review by ttung — APPROVED** [review]

File: src/backend/aspen/app/views/tests/test_phylo_tree_view.py (hunk 3 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/tests/test_phylo_tree_view.py:48]: `aspen/app/views/tests/test_phylo_tree_view.py:48: error: Argument 1 to "sample" has incompatible type "Collection[Sample]"; expected "Union[Sequence[Any], AbstractSet[Any]]"`

**Pull request merged**